gnat2goto/driver/tree_walk.adb

```diff
@@ -1764,6 +1768,57 @@ package body Tree_Walk is
       Do_Type_Declaration (New_Type, Typedef);
    end Do_Itype_Reference;
 
+   -----------------------
+   -- Do_Case_Statement --
+   -----------------------
+
+   function Do_Case_Statement (N : Node_Id) return Irep is
+      Ret : constant Irep := New_Irep (I_Code_Block);
+      Value : constant Irep := Do_Expression (Expression (N));
+
+      --  Auxiliary function to create a single test case
+      --  to emplace in a condition from a list of alternative
+      --  values.
+      function Make_Single_Test (Alt : Node_Id) return Irep;
+      function Make_Single_Test (Alt : Node_Id) return Irep is
+         Ret : constant Irep := New_Irep (I_Op_Eq);
+         Rhs : constant Irep := Do_Expression (Alt);
+      begin
+         Set_Lhs (Ret, Value);
+         Set_Rhs (Ret, Rhs);
+         Set_Type (Ret, New_Irep (I_Bool_Type));
+         return Ret;
+      end Make_Single_Test;
+
+      This_Alt : Node_Id := First (Alternatives (N));
+   begin
+      --  Do-while loop because there must be at least one alternative.
```

Comment:
Maybe I am just old and paranoid BUT ... if you are assuming something it is best to make it an assert rather than a comment; even if it is 'obvious'.

Reply:
Definitely worth asserting, but the grammar does indeed guarantee this is true for now (but again, worth checking anyway, who knows what's going to happen in Ada 2023)

Reply:
Parent link of one Hannes posted (case statements)
https://www.adaic.org/resources/add_content/standards/05rm/html/RM-5-4.html